src/VisualStudio/Core/Def/Implementation/Library/ObjectBrowser/AbstractObjectBrowserLibraryManager.cs

```diff
@@ -16,16 +16,42 @@
 using Microsoft.VisualStudio.OLE.Interop;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
-using Roslyn.Utilities;
 using IServiceProvider = System.IServiceProvider;
 using Task = System.Threading.Tasks.Task;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.Library.ObjectBrowser
 {
     internal abstract partial class AbstractObjectBrowserLibraryManager : AbstractLibraryManager, IDisposable
     {
-        internal readonly VisualStudioWorkspace Workspace;
-        internal readonly ILibraryService LibraryService;
+        internal VisualStudioWorkspace Workspace
+        {
+            get
+            {
+                if (_workspace == null)
+                {
+                    _workspace = _componentModel.Value.GetService<VisualStudioWorkspace>();
+                    _workspace.WorkspaceChanged += OnWorkspaceChanged;
```

Comment:
:memo: This makes me uncomfortable and certainly needs additional scrutiny to ensure race conditions are avoided.

Reply:
If nothing else, it's kinda odd that asking a property will cause a event subscription as a side-effect....

Reply:
Also, do we need to even subscribe at all until somebody has used the object browser or class view?